### PR TITLE
ReportNG Revival

### LINKED ISSUE #1: reportng revival?
reportng is an important stone of the TestNG ecosystem and it is used by many people even if it is not maintained since years.

Some forks are still alive and continue to progress but they don't have the visibility they deserve.

That's why the @testng-team/testng proposes to own its own fork under the `org.testng` groupId on maven central if one or more person(s) want to take the leadership of the project.

@mingyu0519 @ptsiakos77 @AngryTester @guoxn @lenxeon @zhangyy-k @tm93112 @leopold-liu All of you have at least one recent (in the last year) commit. Are you interested in the idea?

### PR DESCRIPTION
**Pull request by email2vimalraj**
Addresses #1 

- Removed unnecessary ANT builds and leverages only maven
- Updated to latest TestNG
- changed from velocity template to freemarker template
- Upgraded the HTML templates to HTML5
- Removed frames as it is deprecated and no more supported from HTML5
- Added navigation menu in the top
- Made some aesthetic changes on the HTML template

**Attached the sample reports**:

_A Simple test output_: [test-output.zip](https://github.com/testng-team/reportng/files/2161629/test-output.zip)
_Suite of Suite test output_: [suite-of-suites-test-output.zip](https://github.com/testng-team/reportng/files/2161628/suite-of-suites-test-output.zip)

**Few screenshots for your reference**:

screencapture-file-users-vimalrajselvam-development-github-reportng-suite-of-suites-test-output-html-index-html-2018-07-04-07_41_31

screencapture-file-users-vimalrajselvam-development-github-reportng-test-output-html-index-html-2018-07-04-07_42_03

screencapture-file-users-vimalrajselvam-development-github-reportng-test-output-html-suite0_test0_results-html-2018-07-04-07_42_22

screencapture-file-users-vimalrajselvam-development-github-reportng-test-output-html-output-html-2018-07-04-07_42_33

screencapture-file-users-vimalrajselvam-development-github-reportng-test-output-html-suite0_test1_results-html-2018-07-04-07_42_53

screencapture-file-users-vimalrajselvam-development-github-reportng-test-output-html-suite0_groups-html-2018-07-04-07_43_07

### CODE REVIEW

**Comment by email2vimalraj** [comment]
ping @juherr and @krmahadevan

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.uncommons</groupId>
```
Comment (krmahadevan) [inline pom.xml]: I dont think TestNG would be able to publish in this space. So we may have to change the group id to `org.testng`

**Review by krmahadevan — CHANGES_REQUESTED** [review]
@email2vimalraj - This is awesome! Thank you for taking my words seriously and helping out with reviving ReportNG. If we can just fix the `groupId` part alone and push in the changes, I think we can get this merged.

@juherr - I was hoping that we just review the pom file and get this changeset in so that we can later iterate on the code as subsequent PRs. There are some changes that we may have to make to the code to fix things etc., but since @email2vimalraj is merely helping port the code to the latest, I think we can skip those and perhaps log bugs etc., for tracking the changes required (minor changes in the codebase). I can work with @email2vimalraj to get those sorted out as well in subsequent PRs. Please let me know your thoughts on this.

File: pom.xml (hunk 1 above)
Comment (email2vimalraj) [reply pom.xml]: Done

**Review by krmahadevan — APPROVED** [review]

**Comment by krmahadevan** [comment]
@juherr - I have got the refactoring also included in this PR. So I guess almost everything is taken care of. Please help approve and merge.

**Review by juherr — APPROVED** [review]

**Comment by krmahadevan** [comment]
@email2vimalraj - Do you think you could help add those screenshots in some good presentable way to the `README.md` as sample outputs for our users to refer to and get a feel of how the reports would look like? It could be a separate PR.

**Comment by email2vimalraj** [comment]
@krmahadevan : Sure will do.

**Pull request merged**